src/basic/os-util.c

```diff
@@ -59,11 +59,13 @@ int open_extension_release(const char *root, const char *extension, char **ret_p
         if (extension) {
                 const char *extension_full_path;
 
-                if (!image_name_is_valid(extension))
+                /* An empty string extension means we should check for the unversioned filename, as a
+                 * fallback for cases where image names are not stable and predictable. */
+                if (!isempty(extension) && !image_name_is_valid(extension))
                         return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
                                                "The extension name %s is invalid.", extension);
 
-                extension_full_path = strjoina("/usr/lib/extension-release.d/extension-release.", extension);
+                extension_full_path = strjoina("/usr/lib/extension-release.d/extension-release", !isempty(extension) ? "." : "", extension);
```

Comment:
but what happens if you have multiple of these? then they will all hog the same file and override each other?

also, I think this would be a log more readable if written like this:

```c
if (isempty(extension))
        extension_full_path = "/usr/lib/extension-release.d/extension-release";
else
        extension_full_path = strjoina(…);
```

i.e. concatenating a constant string with two empty strings is a pointless excercise